packages/rules-engine/src/helpers/previousValueCheck.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import { rulesEngineEffectTargetDataTypes, typeKeys } from '../constants';
+import type { DataElements, TrackedEntityAttributes, HideOutputEffect } from '../rulesEngine.types';
+
+const processDataElementValue = ({
+    dataElementId,
+    dataElements,
+}: {
+    dataElementId: ?string,
+    dataElements: ?DataElements,
+}) => {
+    if (dataElementId && dataElements && dataElements[dataElementId]) {
+        const dataElement = dataElements[dataElementId];
+        return {
+            valueName: dataElement.name,
+            valueType: dataElement.optionSetId ? typeKeys.TEXT : dataElement.valueType,
```

Comment:
We will have to always return `dataElement.valueType` here. The variable service is different because we connvert the value itself to the option set name (and the option set name is always a TEXT)